objects/src/accounts/storage/mod.rs

```diff
@@ -281,11 +290,73 @@ impl AccountStorage {
             return Err(AccountError::StorageSlotIsReserved(index));
         }
 
+        // only value slots of basic arity can currently be updated
+        match self.layout[index as usize] {
+            StorageSlotType::Value { value_arity } => {
+                if value_arity > 0 {
+                    return Err(AccountError::StorageSlotInvalidValueArity {
+                        slot: index,
+                        expected: 0,
+                        actual: value_arity,
+                    });
+                }
+            },
+            StorageSlotType::Map { value_arity } => {
+                if value_arity > 0 {
+                    return Err(AccountError::StorageSlotInvalidValueArity {
+                        slot: index,
+                        expected: 0,
+                        actual: value_arity,
+                    });
+                }
+            },
+            slot_type => Err(AccountError::StorageSlotNotValueSlot(index, slot_type))?,
+        }
+
         // update the slot and return
         let index = LeafIndex::new(index.into()).expect("index is u8 - index within range");
         let slot_value = self.slots.insert(index, value);
         Ok(slot_value)
     }
+
+    /// Updates a storage map at the specified index.
+    ///
+    /// # Errors
+    /// Returns an error if:
+    /// - The index specifies a reserved storage slot.
+    /// - The index is not u8.
+    /// - The map does not exist at the specified index.
+    pub fn set_map_item(
+        &mut self,
+        index: u8,
+        map_delta: StorageMapDelta,
+    ) -> Result<(), AccountError> {
```

Comment:
I think we should keep the signature here as close to MASM as possible. For example, something like this:
```Rust
pub fn set_map_item(
    &mut self,
    index: u8,
    key: Digest,
    value: Word,
) -> Result<(), AccountError>
```

Reply:
Good point. 

We might not have a `value: Word` in the case of a `cleared_map_entry`. So we could leave it as is or we can do an `Option<Word>`. What do you think?

Reply:
For cleared entries the value is actually `EMPTY_WORD` - but as I mentioned in another comment, I think these should be processed via `apply_delta()` method anyway.

Reply:
jup, I process them now in `apply_delta()`